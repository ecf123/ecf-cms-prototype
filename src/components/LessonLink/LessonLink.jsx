import "./LessonLink.scss";
import arrow from "../../assets/arrow-pointing-right-icon.svg";
import playfiles from "../../assets/play-files-icon.svg";
import clock from "../../assets/clock-icon.svg";
import { Link } from "react-router-dom";

export const LessonLink = ({ courseTitle, courseDuration }) => {
  const link = courseTitle?.split(" ").join("");
  return (
    <div className="lesson-link-container">
      <div className="lesson-link-container__course-info">
        <div className="lesson-link-container__title">
          <img src={playfiles} alt="play files icon" />
          <Link style={{ textDecoration: "none" }} to={link}>
            <h5 className="lesson-link-container__title-header">
              {courseTitle}
            </h5>
          </Link>
        </div>
        <div className="lesson-link-container__time">
          <img src={clock} alt="clock icon" />
          <h6 className="lesson-link-container__time-header">
            {courseDuration}
          </h6>
        </div>
      </div>
      <Link to={link}>
        <div className="lesson-link-container__arrow">
          <img
            className="lesson-link-container-arrow-image"
            src={arrow}
            alt="course arrow"
          />
        </div>
      </Link>
    </div>
  );
};

export default LessonLink;
