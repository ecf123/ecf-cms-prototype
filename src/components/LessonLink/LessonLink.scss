@use "../../styles/variables.scss" as *;
@use "../../styles/typography.scss" as *;

.lesson-link-container {
  width: 46.938rem;
  height: 4.938rem;
  display: flex;
  justify-content: space-between;

  &__course-info {
    width: fit-content;
    height: fit-content;
    display: flex;
    flex-direction: column;
  }

  &__title {
    display: flex;
    justify-content: space-between;
    color: $color-teal;
    margin-bottom: 0.438rem;
    gap: 1.323rem;
  }

  &__title-header {
    color: $color-grey-normal;
  }

  &__time {
    display: flex;
    color: $color-grey-normal;
    margin-left: 2.5rem;
    gap: 0.469rem;
  }

  &__time-header {
    color: $color-grey-normal;
  }

  &__arrow {
    height: 3.125rem;
    width: 3.125rem;
    border-radius: 50%;
    background-color: rgba($color-blue-navy-dark, 0.1);
    color: rgba($color-blue-navy-dark, 0.1);
    display: flex;
    margin-top: 0.063rem;
    justify-content: center;
    align-items: center;
  }

  &__arrow-image {
    margin: auto;
  }
}
