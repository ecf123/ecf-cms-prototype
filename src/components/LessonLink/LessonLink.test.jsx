import LessonLink from "./LessonLink";
import { customRender } from "../../utils/testUtils";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

describe("LessonLink component", () => {
  it("matches the snapshot taken", () => {
    const { container } = customRender(
      <LessonLink
        courseDuration={"10 min"}
        courseTitle={"The Role of Banking in the Financial System"}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("should render all images", () => {
    customRender(
      <LessonLink
        courseTitle="How to win a million dollars"
        courseDuration="10 min"
      />,
      { wrapper: BrowserRouter }
    );
    const images = screen.getAllByRole("img");
    expect(images).toBeTruthy();
  });

  it("should render the heading title and course duration", () => {
    render(
      <LessonLink
        courseTitle="How to win a million dollars"
        courseDuration="10 min"
      />,
      { wrapper: BrowserRouter }
    );
    const title = screen.getByText("How to win a million dollars");
    const duration = screen.getByText("10 min");
    expect(title).toBeInTheDocument();
    expect(duration).toBeInTheDocument();
  });
});
