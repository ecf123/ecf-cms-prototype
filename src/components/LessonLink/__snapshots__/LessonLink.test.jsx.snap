// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LessonLink component matches the snapshot taken 1`] = `
<div>
  <div
    class="lesson-link-container"
  >
    <div
      class="lesson-link-container__course-info"
    >
      <div
        class="lesson-link-container__title"
      >
        <img
          alt="play files icon"
          src="play-files-icon.svg"
        />
        <a
          href="/TheRoleofBankingintheFinancialSystem"
          style="text-decoration: none;"
        >
          <h5
            class="lesson-link-container__title-header"
          >
            The Role of Banking in the Financial System
          </h5>
        </a>
      </div>
      <div
        class="lesson-link-container__time"
      >
        <img
          alt="clock icon"
          src="clock-icon.svg"
        />
        <h6
          class="lesson-link-container__time-header"
        >
          10 min
        </h6>
      </div>
    </div>
    <a
      href="/TheRoleofBankingintheFinancialSystem"
    >
      <div
        class="lesson-link-container__arrow"
      >
        <img
          alt="course arrow"
          class="lesson-link-container-arrow-image"
          src="arrow-pointing-right-icon.svg"
        />
      </div>
    </a>
  </div>
</div>
`;
